packages/core/src/oidc/utils.test.ts

```diff
@@ -1,21 +1,85 @@
-import { ApplicationType } from '@logto/schemas';
+import { ApplicationType, CustomClientMetadataType } from '@logto/schemas';
 
-import { getApplicationTypeString, buildOidcClientMetadata } from './utils';
+import {
+  allowOrigin,
+  buildOidcClientMetadata,
+  getApplicationTypeString,
+  validateCustomClientMetadata,
+} from './utils';
 
-describe('oidc utils method', () => {
-  it('getApplicationTypeString', () => {
-    expect(getApplicationTypeString(ApplicationType.SPA)).toEqual('web');
-    expect(getApplicationTypeString(ApplicationType.Native)).toEqual('native');
-    expect(getApplicationTypeString(ApplicationType.Traditional)).toEqual('web');
+it('getApplicationTypeString', () => {
+  expect(getApplicationTypeString(ApplicationType.SPA)).toEqual('web');
+  expect(getApplicationTypeString(ApplicationType.Native)).toEqual('native');
+  expect(getApplicationTypeString(ApplicationType.Traditional)).toEqual('web');
+});
+
+it('buildOidcClientMetadata', () => {
+  const metadata = {
+    redirectUris: ['logto.dev'],
+    postLogoutRedirectUris: ['logto.dev'],
+    logoUri: 'logto.pnf',
+  };
+  expect(buildOidcClientMetadata()).toEqual({ redirectUris: [], postLogoutRedirectUris: [] });
+  expect(buildOidcClientMetadata(metadata)).toEqual(metadata);
+});
+
+describe('validateMetadata', () => {
+  describe('corsAllowedOrigins', () => {
+    it('should not when allowed origins are all valid', () => {
+      expect(() => {
+        validateCustomClientMetadata('corsAllowedOrigins', [
+          'http://localhost:3001',
+          'https://logto.dev',
+        ]);
+      }).not.toThrow();
+    });
+
+    it('should throw when allowed origins are not all valid', () => {
+      expect(() => {
+        validateCustomClientMetadata('corsAllowedOrigins', ['', 'logto.dev/sign-in']);
+      }).toThrow();
+    });
+  });
+
+  // Describe.each(['idTokenTtl', 'refreshTokenTtl'])('%s', (ttlKey) => {
+  //   test(`${ttlKey} should not throw when it is a number`, () => {
+  //     expect(() => {
+  //       validateCustomClientMetadata(ttlKey, 5000);
+  //     }).not.toThrow();
+  //   });
+  //
+  //   test(`${ttlKey} should throw when it is not a number`, () => {
+  //     expect(() => {
+  //       validateCustomClientMetadata(ttlKey, 'string_value');
+  //     }).toThrow();
+  //   });
```

Comment:
wonder why it does not throw when idTokenTtl or refreshTokenTtl is not a number……

![image](https://user-images.githubusercontent.com/10594507/162209812-0bb5f821-d221-48ed-980d-74b40b12f749.png)

Reply:
@simeng-li I'm confused that the original code does not work as well.
e.g. whether idTokenTtl is a number or a string, `guard.safeParse` always return `result === true`
![image](https://user-images.githubusercontent.com/10594507/162210514-b70c0105-fe7c-4b2e-9699-d5a22e5870b8.png)

Reply:
![image](https://user-images.githubusercontent.com/10594507/162214392-4c47fbcb-9621-4dff-b49b-68114fe16cf0.png)

@logto-io/eng

Reply:
Get it.

- Correct usage:
    ![image](https://user-images.githubusercontent.com/10594507/162215468-75dd5dd4-c1b3-4819-84a1-53e1edcf44ff.png)

- Wrong usage:
    ![image](https://user-images.githubusercontent.com/10594507/162215595-c646bfbe-8b13-4f41-83c1-9ea79a3eeeb8.png)